src/transformers/generation/logits_process.py

```diff
@@ -392,17 +392,57 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
 
 class TypicalLogitsWarper(LogitsWarper):
     r"""
-    [`LogitsWarper`] that performs typical decoding. See [Typical Decoding for Natural Language
-    Generation](https://arxiv.org/abs/2202.00666) for more information.
+    [`LogitsWarper`] that enforces the criterion of locally typical sampling when generating from probabilistic language models.
+     The abstraction of natural language generation as a discrete stochastic process, analyzed through information theory,
+    provides insights into the behavior of probabilistic language generators. This warping technique aims to produce more
+    coherent and fluent text by selecting words that have information content close to the expected information content,
+    i.e., the conditional entropy of the model. See [Typical Decoding for Natural Language Generation](https://arxiv.org/abs/2202.00666) for more information.
 
+    <Tip>
+
+     The `TypicalLogitsWarper` is a special case of a `LogitsWarper`, which is itself a subclass of `LogitsProcessor`.
+        It is designed to modify the logits during text generation with probabilistic language models when `do_sample=True`.
+        By using `TypicalLogitsWarper`, you can enforce locally typical sampling, resulting in more coherent and fluent text,
+        while reducing degenerate repetitions.
+
+    </Tip>
```

Comment:
This `tip` could be summarized as a single short sentence, where it would point to the need to pass `do_sample=True` (without repeating what typical sampling is for, as it is already stated in at the top)